messaging/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -1042,7 +1042,11 @@ public void run() {
                 int[] tokenStoreCurrentSegments;
 
                 try {
-                    tokenStoreCurrentSegments = tokenStore.fetchSegments(processorName);
+                    tokenStoreCurrentSegments = transactionManager.fetchInTransaction(
+                            () -> {
```

Comment:
Can't this be simplified to `() -> tokenStore.fetchSegments(processorName)`? I'd guess Intellij actually "warns" you about this.